src/zeep/xsd/elements.py

```diff
@@ -76,9 +76,13 @@ def __call__(self, *args, **kwargs):
         return self.type(*args, **kwargs)
 
     def _signature(self, name):
+        if hasattr(self.type, 'name'):
+            typename = self.type.name
+        else:
+            typename = self.type.__repr__
```

Comment:
This could actually be written as `typename = getattr(self.type, 'name', repr(self.type))`